src/main/java/com/artipie/api/RepositoryRest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.api;
+
+import com.artipie.misc.JavaResource;
+import com.artipie.settings.repo.CrudRepoSettings;
+import io.vertx.core.json.JsonArray;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.FileSystemAccess;
+import io.vertx.ext.web.handler.StaticHandler;
+import io.vertx.ext.web.openapi.RouterBuilder;
+import java.io.StringReader;
+import javax.json.JsonObject;
+import javax.json.JsonStructure;
+import javax.json.spi.JsonProvider;
+import org.eclipse.jetty.http.HttpStatus;
+
+/**
+ * Rest-api operations for repositories settings CRUD
+ * (create/read/update/delete) operations.
+ * @since 0.26
+ */
+@SuppressWarnings({"PMD.OnlyOneReturn", "PMD.AvoidDuplicateLiterals"})
+public final class RepositoryRest extends BaseRest {
+    /**
+     * Repository settings create/read/update/delete.
+     */
+    private final CrudRepoSettings crs;
+
+    /**
+     * Artipie layout.
+     */
+    private final String layout;
+
+    /**
+     * Ctor.
+     * @param crs Repository settings create/read/update/delete
+     * @param layout Artipie layout
+     */
+    public RepositoryRest(final CrudRepoSettings crs, final String layout) {
+        this.crs = crs;
+        this.layout = layout;
+    }
+
+    @Override
+    public void init(final RouterBuilder rbr) {
+        rbr.operation("listAll")
+            .handler(this::listAll)
+            .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        if ("flat".equals(this.layout)) {
+            rbr.operation("getRepo")
+                .handler(this::getRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("createRepo")
+                .handler(this::createRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        } else {
+            rbr.operation("list")
+                .handler(this::listUserRepos)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("getUserRepo")
+                .handler(this::getRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("createUserRepo")
+                .handler(this::createRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        }
+        final Router router = rbr.createRouter();
+        router.route("/api/*").handler(
+            StaticHandler.create(
+                FileSystemAccess.ROOT,
+                new JavaResource("swagger-ui").uri().getPath()
+            )
+        );
```

Comment:
Do we need to expose swagger here? As far as I understand, we do in `RestApi` class with correct index

Reply:
All content of swagger-ui folder is exposed outside including yaml-folder and index-org.html, index-flat.html

Reply:
RestApi is up the rest controller that build based on yaml open api spec.
But swagger-ui should be exposed because it shows yaml in swagger UI.